### PR TITLE
refactor(api): remove rx from `boot/user.js`

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pul-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Reduces complexity by removing _the dreadfully powerful `rx`_

### CODE REVIEW

**Review by raisedadead — COMMENTED** [review]
Thanks for refactoring this. 

We should also look at the actual queries that are fired to the DB.

Here are some thoughts:

1. If we are firing 5 promises, could those be straightened/optimized, so they are doing better? Could we do fewer/DRY?

2. Why would the request not have the `user` object in the format we want? Let's say loopback wants a `username` or other props on the thing. Should it be pre-filled (either from the client or just using the instance method(s)?

**Comment by ShaunSHamilton** [comment]
> 1. f we are firing 5 promises, could those be straightened/optimized, so they are doing better? Could we do fewer/DRY?

Ideally, we would forgo everything in `common/models/`, and just supply the connection to Mongodb. Then, we could make each request as dry as possible - `collection.findOne(filter, projection)`

Do you want this within this PR?

---

> 2. Why would the request not have the user object in the format we want? Let's say loopback wants a username or other props on the thing. Should it be pre-filled (either from the client or just using the instance method(s)?

I am not sure I fully understand your point/question.

If you are referring to the error (`username is a required argument`), then I am still unsure where this is going wrong. Loopback has far too many internals which are configured through random/magic strings (e.g. all of the middleware lifecycle).

Again, **we** can do what we like with objects passed back and forth (instance methods). Loopback does its own thing, and validates this is done at unbeknownst-to-me stages.

The client should never pre-fill things unless they are public - it is a waste of time for the client to request with _"I am username `x`, and want to get my user data"_.

**Comment by raisedadead** [comment]
You can't just sidestep the ACL foundations from Loopback and directly talk to MongoDB at will like that. 

**This PR is OK, and I get this is not a fix for the existential issues. We can merge it.**

The bug is because we are not correctly using the "remote method" (a.k.a instance method) paradigm in the Loopback world. 

So, it works like this:

Loopback's job is to enforce type safety and access control (ACL). You can think of remote methods as RPC-style methods that can be executed on the backend when a request is sent to it.

When a request is made to an endpoint defined as an instance method in a model, say, User, the requestee passes the "user" object along with authorization (Loopback-based access token).

It's LB's job to verify that the "context" of the request is legit. That is the user object in the request has the access token and can execute queries. 

These queries are then translated to MongoDB transactions, and LB becomes our ORM and HTTP-based REST API backend.

What I suspect is happening is a mismatch/overlap in the user object sent from the client and that LB expects.

This "context" thing makes and breaks it. You can learn more about this here: https://loopback.io/doc/en/lb3/Remote-methods.html

But the basic principle is that:

So while a typical query for a direct connection might look like so: 

```
const UserCol = db.name_of_the_user_collection;
const filter = { 
	email: { $exists: true }, 
	username: { $exists: true } 
}
const user = UserCol.findOne(filter, ...);
...
```

Above you do not have a concrete user object (as in a JavaScript object that is an instance of the "User" class).

In the LB world, it becomes:

```
const user = req.user; // here a user is a concrete object (an instance of the User Class enforced by LB ACL framework"
user.getMyDataMentod();
```

Here, we define the `getMyDataMenthod` and an actual method in the User Class. Some methods like `user.login` are defined by LB (there are defaults, and we extend them).

I want to understand then,

Why is the user instance malformed if at all? Who is doing that? Are we modifying the LB's instance of the user object unintentionally?

**Comment by raisedadead** [comment]
My point is - if we are unintentionally malformed the instance with LB's protections in place, we can make the situation worse in the new API with all the type safety around the object relaxed with our instance definition, etc.

Tl;DR: 

We will be reinventing the wheel sort of, in the new API, and we could invent it as an oval if not a cube, thinking oh the wheel is definitely a circle.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:
